zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -519,6 +519,21 @@ public void setPropertyAndRestart(String id, InterpreterOption option,
     synchronized (interpreterSettings) {
       InterpreterSetting intpsetting = interpreterSettings.get(id);
       if (intpsetting != null) {
+
```

Comment:
It would be better to refactor setPropertyAndRestart() and restart() to remove duplication and prevent diverged issue like this.